challenge_4/x86/x95/src/invertTree.s

```diff
@@ -0,0 +1,62 @@
+.text
+.global invertTree
+
+invertTree:
+	#x86 Prolog
+	pushq %rbp
+	movq %rsp, %rbp
+
+	# %rdi is input according to AMD64
+	xorq %rsi,%rsi 	# %rsi is used to read/write nodes from memory
+	xorq %r8,%r8	# %r8 saves the address of the right node
+	xorq %r9,%r9	# %r9 saves the address of the left node
+
+	# Get the right node
+	movq %rdi,%rsi
+	add $8, %rsi
+	movq (%rsi),%rsi
+
+	# Traverse right node first
+	cmp $0,%rsi
+	je skip_right
+	pushq %rdi	# Preserve %rdi and proceed with %rsi
+	movq %rsi,%rdi
+	call invertTree
+	popq %rdi
+	skip_right:
+
+	# Get the left node
+	movq %rdi,%rsi
+	add $16, %rsi
+	movq (%rsi),%rsi
+
+	# Traverse left node
+	cmp $0,%rsi
+	je skip_left
+	pushq %rdi	# Preserve %rdi and proceed with %rsi
+	movq %rsi,%rdi
+	call invertTree
+	popq %rdi
+	skip_left:
+
+	# Get addresses of both nodes
```

Comment:
I know you use linux. It was I who didn't pay attention and sat here wondering why it wasn't working, lol.

No worries, always good to know other ways to do things.  :)